Plugins~/Src/MeshSyncClientBlender/msblenContextIntermediatePathProvider.cpp

```diff
@@ -2,9 +2,33 @@
 #include "msblenUtils.h"
 #include <BLI_listbase.h>
 
-std::string append_id(std::string path, const Object* obj) {
+std::string msblenContextIntermediatePathProvider::append_id(std::string path, const Object* obj) {
     auto data = (ID*)obj->data;
-    path += "_" + std::to_string(data->session_uuid);
+    
+    path += "_" + std::string(data->name);
+
+    // If we already have an object with this name but a different session_uuid, append the session_uuid as well
+    auto it = mappedNames.find(data->name);
+    
+    if (it == mappedNames.end()) {
+        mappedNames.insert(std::make_pair(data->name, data->session_uuid));
+    }
+    else {
+        bool found = false;
+        for (; it != mappedNames.end(); it++)
+        {
+            if (it->second == data->session_uuid) {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            path += "_" + std::to_string(data->session_uuid);
+            mappedNames.insert(std::make_pair(data->name, data->session_uuid));
+        }
+    }
```

Comment:
Why not use an unordered set tracking the session_uuids here?

Reply:
I only want to append a session id if I already have an item with the same name. I would not know the session ids of other items with this name here without the mappedNames collection.

Reply:
is the session id unique across all items or unique across all items with the same name?

Reply:
it's unique across all items

Reply:
Another concern is the scope and ownership of the cache. As it stands, if we have multiple geonode trees, we will get cache hits only for the tree that was executed first. My suggestion is to extract this into a cache object that is owned by the blender context class (or, in the future ,by the appropriate exported/handler class for geonodes). The owner can use the cache to decide which provider to use. This will keep this provider simple with a single responsibility (the caching part is not reflected anywhere in the name or comments).